Calculator created

DiLevchenko opened the pull request.

DiLevchenko committed:
Calculator created

DiLevchenko committed:
Modulo operator added.
User dialogue message order fixed.

DiLevchenko committed:
Module operator added.
User dialogue message order fixed.

DiLevchenko committed:
Merge remote-tracking branch 'origin/task2_create_a_calculator' into task2_create_a_calculator

Hmeliumm commented on projectJava/task2_calculator/src/Calculator_DL.java:
```diff
@@ -0,0 +1,46 @@
+import java.util.Scanner;
+
+public class Calculator_DL {
```
Snake case (_) is bad practice. Please change to CamelCase

Hmeliumm commented on projectJava/task2_calculator/src/Calculator_DL.java:
```diff
@@ -0,0 +1,46 @@
+        char operator;
+        double num1, num2, result;
+        Scanner input = new Scanner(System.in);
+
```
Please remove extra line

Hmeliumm commented on projectJava/task2_calculator/src/Calculator_DL.java:
```diff
@@ -0,0 +1,46 @@
+        input.close();
+    }
+}
+
```
Please remove extra line.
Format code (Ctrl + Alt+L)

DiLevchenko committed:
Calculation extracted as executeCalculation method.
Now calculator is looped to give user opportunity to repeat calculation, if needed.
try...catch added to handle incorrect number value input.
Class Calculator_DL renamed to CalculatorDL, according to Java convention .

DiLevchenko committed:
Calculation extracted as executeCalculation method.
Now calculator is looped to give user opportunity to repeat calculation, if needed.
try...catch added to handle incorrect number value input.
Class Calculator_DL renamed to CalculatorDL, according to Java convention .

Hmeliumm commented on projectJava/task2_calculator/src/CalculatorDL.java:
```diff
@@ -0,0 +1,61 @@
+    static void executeCalculation(double num1, double num2, char operator) {
+        double result;
+        switch (operator) {
+            case '+':
```
Please move arithmetic operators to constant, in case you are using it more than one time (here and below)

Hmeliumm commented on projectJava/task2_calculator/src/CalculatorDL.java:
```diff
@@ -0,0 +1,61 @@
+        switch (operator) {
+            case '+':
+                result = num1 + num2;
+                System.out.println(num1 + " + " + num2 + " = " + result);
```
Please try to use printf for such cases: System.out.printf(%f + %f + %f, num1, num2, result) (here and below)

Hmeliumm commented on projectJava/task2_calculator/src/CalculatorDL.java:
```diff
@@ -0,0 +1,61 @@
+                executeCalculation(num1, num2, operator);
+
+                System.out.println("To do new calculation enter anything but \'c\'.");
+                repeatLoop = input.next().charAt(0) != 'c';
```
Please move this condition to while

Hmeliumm commented on projectJava/task2_calculator/src/CalculatorDL.java:
```diff
@@ -0,0 +1,61 @@
+                break;
+            }
+        } while (repeatLoop);
+        input.close();
```
Good job!

DiLevchenko replied:
Thanks)

DiLevchenko committed:
Add const. Update  System.out.println to Printf.

Hmeliumm commented on projectJava/task2_calculator/src/CalculatorDL.java:
```diff
@@ -0,0 +1,65 @@
+public class CalculatorDL {
+    static void executeCalculation(double num1, double num2, char operator) {
+        double result;
+        final char add = '+';
```
Please remember that we write constants all in uppercase (like ADD and so on)

Hmeliumm commented on projectJava/task2_calculator/src/CalculatorDL.java:
```diff
@@ -0,0 +1,65 @@
+
+                System.out.println("To do new calculation enter anything but \'c\'.");
+            } catch (Exception e) {
+                System.out.println(e);
```
If you catch Exception then print stack trace at least in this point

Hmeliumm commented on projectJava/task2_calculator/src/CalculatorDL.java:
```diff
@@ -0,0 +1,65 @@
+        input.close();
+    }
+}
+
```
Remove extra lines

DiLevchenko committed:
Rewrote constants in uppercase. Add a stack trace. Remove extra line.

DiLevchenko committed:
Remove break.
